ChatBot/ChatBot/Presentation/GPTChatRooms/ViewController/GPTChatRoomsViewController.swift

```diff
@@ -0,0 +1,154 @@
+        let input = GPTChatRoomsInput(
+            fetchRooms: fetchRoomsSubject.eraseToAnyPublisher(),
+            createRoom: createRoomSubject.eraseToAnyPublisher(),
+            modifyRoom: modifyRoomSubject.eraseToAnyPublisher(),
+            deleteRoom: deleteRoomSubject.eraseToAnyPublisher(),
+            selectRoom: selectRoomSubject.eraseToAnyPublisher()
+        )
+        
+        viewModel.transform(from: input)
+            .sink { [weak self] output in
+                switch output {
+                case .success(let rooms):
+                    self?.applySnapshot(rooms: rooms)
+                case .failure(let error):
+                    self?.present(UIAlertController(error: error), animated: true)
+                case .moveToChatRoom(let chatRoomViewModel):
+                    let viewController = GPTChattingViewController(viewModel: chatRoomViewModel)
+                    self?.navigationController?.pushViewController(viewController, animated: true)
+                }
+            }
+            .store(in: &cancellables)
+    }
+}
+
+// MARK: - Configure UI
+extension GPTChatRoomsViewController {
+    private func configureUI() {
+        navigationItem.title = "MyChatBot 🤖"
+        navigationItem.rightBarButtonItem = UIBarButtonItem(image: UIImage(systemName: "pencil"), style: .plain, target: self, action: #selector(createChatRoomButtonTapped(_:)))
+    }
+}
+
+// MARK: - Confiure TableView
+extension GPTChatRoomsViewController {
+    private func configureDataSource() {
+        self.chatRoomsDataSource = UITableViewDiffableDataSource(tableView: tableView) { tableView, indexPath, itemIdentifier in
+            let cell = tableView.dequeueReusableCell(withIdentifier: "GPTChatRoomCell", for: indexPath)
+            var config = cell.defaultContentConfiguration()
+            config.text = itemIdentifier.title
+            config.secondaryText = itemIdentifier.recentChatDate.toLocaleString(.koKR)
+            cell.contentConfiguration = config
+            return cell
+        }
+    }
+    
+    private func applySnapshot(rooms: [Model.GPTChatRoomDTO]) {
+        var snapshot = NSDiffableDataSourceSnapshot<Section, Model.GPTChatRoomDTO>()
+        snapshot.appendSections([.main])
+        snapshot.appendItems(rooms)
+        chatRoomsDataSource.apply(snapshot)
+    }
+}
+
+extension GPTChatRoomsViewController {
+    @objc
+    private func createChatRoomButtonTapped(_ Sender: Any) {
+        present(configureCreateRoomAlert(), animated: true)
+    }
+}
+
+extension GPTChatRoomsViewController {
+    private func configureCreateRoomAlert() -> UIAlertController {
+        UIAlertController(title: "방 생성", message: "생성할 방 제목을 입력해주세요.") { [unowned self] roomName in
+            createRoomSubject.send(roomName)
+        }
+    }
+    
+    private func configureModifyRoomAlert(for indexPath: IndexPath) -> UIAlertController {
+        UIAlertController(title: "방 수정", message: "수정할 방 제목을 입력해주세요.") { [unowned self] roomName in
```

Comment:
```suggestion
        UIAlertController(title: "방 수정", message: "수정할 방 제목을 입력해주세요.") { [weak self] roomName in
```

가능하다면 publisher 인터페이스 제공을 희망함